#! /bin/sh -e

# DP: use -fwrapv when GCC supports it.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        rm -f configure
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- configure.in~	2006-10-17 18:03:36.000000000 +0200
+++ configure.in	2007-12-25 12:12:55.000000000 +0100
@@ -735,6 +735,10 @@
 then
     case $GCC in
     yes)
+        # For gcc 4.x we need to use -fwrapv so lets check if its supported
+        if "$CC" -v --help 2>/dev/null |grep -- -fwrapv > /dev/null; then
+           WRAP="-fwrapv"
+        fi
 	case $ac_cv_prog_cc_g in
 	yes)
 	    if test "$Py_DEBUG" = 'true' ; then
@@ -742,11 +746,11 @@
 		# debug builds.
 		OPT="-g -Wall -Wstrict-prototypes"
 	    else
-		OPT="-g -O3 -Wall -Wstrict-prototypes"
+		OPT="-g $WRAP -O3 -Wall -Wstrict-prototypes"
 	    fi
 	    ;;
 	*)
-	    OPT="-O3 -Wall -Wstrict-prototypes"
+	    OPT="$WRAP -O3 -Wall -Wstrict-prototypes"
 	    ;;
 	esac
 	case $ac_sys_system in
